src/samv71_serial_ccsds/samv71_serial_ccsds.c

```diff
@@ -276,26 +280,24 @@ void SamV71SerialCcsdsPoll(void *private_data) {
 
   Escaper_start_decoder(&self->m_escaper);
   xSemaphoreTake(self->m_rx_semaphore, portMAX_DELAY);
+
   Hal_uart_read(&self->m_hal_uart, self->m_fifo_memory_block,
                 Serial_CCSDS_SAMV71_RECV_BUFFER_SIZE, self->m_uart_rx_handler);
 
   while (true) {
     /// Wait for data to arrive. Semaphore will be given
     xSemaphoreTake(self->m_rx_semaphore, portMAX_DELAY);
+
     length = ByteFifo_getCount(&self->m_hal_uart.rxFifo);
-    if (length <= 0) {
-      Hal_console_usart_write(
-          (const uint8_t *const)SAMV71_SERIAL_CCSDS_POOL_ERROR,
-          sizeof(SAMV71_SERIAL_CCSDS_POOL_ERROR));
-      assert(false && SAMV71_SERIAL_CCSDS_POOL_ERROR);
-      return;
-    } else {
-      for (size_t i = 0; i < length; i++) {
-        ByteFifo_pull(&self->m_hal_uart.rxFifo, &self->m_recv_buffer[i]);
-      }
-      Escaper_decode_packet(&self->m_escaper, self->m_recv_buffer, length,
-                            Broker_receive_packet);
+
+    for (size_t i = 0; i < length; i++) {
+      self->m_hal_uart.uart.reg->idr = UART_IDR_RXRDY_MASK;
```

Comment:
I assume this is an interrupt disable. Don't we have a function for that?

Reply:
We could also resign from passing a reference to error code at all. Why not handle all of the errors in the error handler.

Reply:
Ok, there was a private inline function on the main branch. It also has a meaningful name. I will use that function.